core/arch/x86/instr.c

```diff
@@ -343,7 +343,8 @@ instr_is_mov(instr_t *instr)
             opc == OP_mov_ld ||
             opc == OP_mov_imm ||
             opc == OP_mov_seg ||
-            opc == OP_mov_priv);
+            opc == OP_mov_priv ||
+            opc == OP_movzx);
```

Comment:
There are several issues here:

1) If we were to expand what instr_is_mov() covers, then its header documentation, which today explicitly lists every opcode it includes, would have to be updated in this patch.  We do not want the code not matching the docs.

2) I'm not sure we want to expand what it covers: as the docs say, it's supposed to combine the opcodes we've separated in our OP_ enum back into the assembler mnemonic "mov".  "movzx" is a distinct mnemonic.  This is not supposed to cover "any instruction that moves data".

3) There are some callers of this routine in our own code, and maybe in some external users' code: this might break them as they may rely on this covering only the current opcodes it covers.  Did you check the uses in the DR repo?

Reply:
1. Of course for any change the corresponding docs would have to be updated.

2. The question here is if movzx can be regarded as a mov instruction. From a semantics of the instruction as well as its name one would argue that indeed it does, however as you say it is a distinct mnemonic. I am not sure this patch would make such big difference since the current opcodes are defined clearly in the documentation and one can easily add a an extra check for  movzx and even movsx. This is just from a usability standpoint.

3. I could only find instr_is_mov being used in the annotations module under core/annotations.c for the IS_ANNOTATION_LABEL_INSTR macro. I am not familiar with the module but it appears to me that an instruction like movzx would not be used for annotations. From my understanding the relevant code would ignore movzx since it would not correspond to annotation code so it shouldn't break anything.